File: vue-spotify-app.client/src/classes/trackViewModelBatch.ts
import type TrackViewModel from "./trackViewModel";
// Represents a batch of track view models, used to help with virtual scrolling in the frontend.
export default class TrackViewModelBatch {
  // The index of the batch records found.
  batchIndex: number = 0;
  // Represents the tracks found in the batch.
  public trackViewModels: TrackViewModel[] = [];
}

File: vue-spotify-app.Classes/TrackViewModelBatch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace vue_spotify_app.Classes
{
    /// <summary>
    /// Represents a batch of track view models, used to help with virtual scrolling in the frontend.
    /// </summary>
    public class TrackViewModelBatch
    {
        /// <summary>
        /// The index of the batch records found.
        /// </summary>
        public int BatchIndex { get; set; }
        /// <summary>
        /// Represents the tracks found in the batch.
        /// </summary>
        public List<TrackViewModel> TrackViewModels { get; set; } = new List<TrackViewModel>();
    }
}
